teachertool/src/utils/index.ts

```diff
@@ -1,5 +1,6 @@
 import { nanoid } from "nanoid";
 import { NotificationWithId } from "../types";
+import { stateAndDispatch } from "../state";
```

Comment:
Let's keep app state out of the utils file. I suggest moving `getCatalogCriteriaWithId` to /src/state/helpers.ts. We'll probably accumulate more of these helpers, and I'd like to keep utils uncomplicated.